code/game/machinery/bees_apiary.dm

```diff
@@ -103,180 +178,370 @@
-			if(prob(10))
-				toxic = max(0, toxic - 1)
-
-		//handle nutrients
-		nutrilevel -= bees_in_hive / 10 + owned_bee_swarms.len / 5
-		if(nutrilevel > 0)
-			bees_in_hive += 1 * yieldmod
-			health = min(health+1,maxhealth)
-		else
-			//nutrilevel is less than 1, so we're effectively subtracting here
-			health += max(nutrilevel - 1, round(-health / 2))
-			bees_in_hive += max(nutrilevel - 1, round(-bees_in_hive / 2))
-			if(owned_bee_swarms.len)
-				var/mob/living/simple_animal/bee/B = pick(owned_bee_swarms)
-				B.target_turf = get_turf(src)
-
-		//clear out some toxins
-		if(toxic > 0)
-			toxic -= 1
-			health -= 1
-
-		var/newlevel = 0
-		if(health >= (HONEYCOMB_COST * 6))
-			newlevel = 3
-		else if(health >= (HONEYCOMB_COST * 3))
-			newlevel = 2
-		else if(health >= HONEYCOMB_COST)
-			newlevel = 1
-		else
-			newlevel = 0
+		//HANDLE BEEZEEZ
+		if(beezeez)
+			beezeez--
+			nutrilevel++
 
-		if(newlevel != honey_level)
-			overlays -= image('icons/obj/apiary_bees_etc.dmi', icon_state="apiary[honey_level]")
-			overlays += image('icons/obj/apiary_bees_etc.dmi', icon_state="apiary[newlevel]")
-			honey_level = newlevel
+			if(toxic > 0)
+				toxic--
 
-		if(health <= 0)
-			return
 
+		//HANDLE NUTRILEVEL
+		nutrilevel -= worker_bees_inside / 20 + queen_bees_inside /5 + bees_outside_hive.len / 10 //Bees doing work need more nutrients
+
+		nutrilevel += 5 * reagents.trans_to(consume, reagents.total_volume / 100)
+
+
+		for(var/datum/reagent/R in reagents.reagent_list)
+			if (R.volume < 0.01)
+				if (R == NUTRIMENT) continue
+				reagents.del_reagent(R.id,update_totals=0)
+
+		nutrilevel = min(max(nutrilevel,-10),100)
+
+		//PRODUCING QUEEN BEES
+		if(reagents.get_reagent_amount(ROYALJELLY) >= 0.5 && nutrilevel > 10 && queen_bees_inside <= 0 && worker_bees_inside > 1)
+			queen_bees_inside++
+			reagents.remove_reagent(ROYALJELLY, 0.5)
+			worker_bees_inside--
+
+
+		//PRODUCING WORKER BEES
+		if(nutrilevel > 10 && queen_bees_inside > 0 && worker_bees_inside < 20)
+			worker_bees_inside += queen_bees_inside
+
+		// We're getting in dire need of nutrients, let's starve bees so others can survive
+		else if (nutrilevel < -5 && worker_bees_inside >= 10)
+			nutrilevel += 3
+			worker_bees_inside--
+			new/obj/effect/decal/cleanable/bee(get_turf(src))
+
+		//We're low on nutrients, let's call back some bees to reduce our food costs
+		else if (nutrilevel <= 0 && bees_outside_hive.len > 1)
+			for (var/i = 1 to max(1,round(bees_outside_hive.len/3)))
+				var/datum/bee/B = pick(bees_outside_hive)
+				B.homeCall()
+
+
+		//HANDLE TOXICITY
+		var/list/toxic_reagents = list(
+			TOXIN = 2,
+			STOXIN = 1,
+			FLUORINE = 1,
+			RADIUM = 3,
+			FUEL = 2,
+			VOMIT = 1,
+			BLEACH = 2,
+			PLANTBGONE = 3,
+			PLASMA = 2,
+			SACID = 1,
+			PACID = 3,
+			CYANIDE = 4,
+			AMATOXIN = 2,
+			AMANATIN = 3,
+			POISONBERRYJUICE = 2,
+			CARPOTOXIN = 2,
+			ZOMBIEPOWDER = 3,
+			MINDBREAKER = 1,
+			PLASTICIDE = 2,
+		)
```

Comment:
Add a note to reagents so that people update this list when they add a new toxin.